packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,623 @@
+import {
+  Diagnostic,
+  DiagnosticCollector,
+  NoTarget,
+  SourceFile,
+  createDiagnosticCollector,
+  resolvePath,
+} from "@typespec/compiler";
+import { HttpStatusCodeRange } from "@typespec/http";
+import { resolveOperationId } from "@typespec/openapi";
+import {
+  SdkAnyExample,
+  SdkArrayExample,
+  SdkArrayType,
+  SdkBodyModelPropertyType,
+  SdkClientType,
+  SdkDictionaryExample,
+  SdkDictionaryType,
+  SdkHttpOperation,
+  SdkHttpOperationExample,
+  SdkHttpParameter,
+  SdkHttpParameterExample,
+  SdkHttpResponse,
+  SdkHttpResponseExample,
+  SdkModelExample,
+  SdkModelPropertyType,
+  SdkModelType,
+  SdkNullExample,
+  SdkServiceMethod,
+  SdkServiceOperation,
+  SdkType,
+  SdkTypeExample,
+  SdkUnionExample,
+  isSdkFloatKind,
+  isSdkIntKind,
+} from "./interfaces.js";
+import { TCGCContext, getValidApiVersion } from "./internal-utils.js";
+import { createDiagnostic } from "./lib.js";
+
+interface LoadedExample {
+  readonly relativePath: string;
+  readonly file: SourceFile;
+  readonly data: any;
+}
+
+async function loadExamples<TServiceOperation extends SdkServiceOperation>(
+  context: TCGCContext,
+  client: SdkClientType<TServiceOperation>
+): Promise<[Map<string, Record<string, LoadedExample>>, readonly Diagnostic[]]> {
```

Comment:
well it is confusing if an api is returning a map.
How about we return an array of something?
Same comment to the `Record<string, LoadedExample>>`

Reply:
the map is for the efficiency for finding the examples of an operation. since it is not exported, i just added some doc for it.